src/Games/NexusMods.Games.RedEngine/Cyberpunk2077/SortOrder/RedMod/RedModSortableItemProvider.cs

```diff
@@ -0,0 +1,387 @@
+    private readonly SortOrderId _sortOrderId;
+    private readonly SemaphoreSlim _semaphore = new(1, 1);
+    private readonly CompositeDisposable _disposables = new();
+
+    public ReadOnlyObservableCollection<ISortableItem> SortableItems => _readOnlyOrderList;
+
+
+    public LoadoutId LoadoutId { get; }
+    public ISortableItemProviderFactory ParentFactory { get; }
+
+    public static async Task<RedModSortableItemProvider> CreateAsync(
+        IConnection connection,
+        LoadoutId loadoutId,
+        ISortableItemProviderFactory parentFactory)
+    {
+        var sortOrderModel = await GetOrAddSortOrderModel(connection, loadoutId, parentFactory);
+        return new RedModSortableItemProvider(connection,
+            loadoutId,
+            sortOrderModel,
+            parentFactory
+        );
+    }
+
+    private RedModSortableItemProvider(
+        IConnection connection,
+        LoadoutId loadoutId,
+        RedModSortOrder.ReadOnly sortOrderModel,
+        ISortableItemProviderFactory parentFactory)
+    {
+        _connection = connection;
+        LoadoutId = loadoutId;
+        ParentFactory = parentFactory;
+        _sortOrderId = sortOrderModel.AsSortOrder().SortOrderId;
+
+        // load the previously saved order
+        var order = RetrieveSortOrder();
+        _orderCache.AddOrUpdate(order);
+
+        _orderCache.Connect()
+            .Transform(item => item as ISortableItem)
+            .SortBy(item => item.SortIndex)
+            .Bind(out _readOnlyOrderList)
+            .Subscribe()
+            .AddTo(_disposables);
+
+
+        // Observe changes in the RedMods and adjust the order list accordingly
+        RedModLoadoutGroup.ObserveAll(_connection)
+            .Filter(group => group.AsLoadoutItemGroup().AsLoadoutItem().LoadoutId == LoadoutId)
+            .SortBy(g => RedModFolder(g).ToString())
+            .Bind(out _redModsGroups)
+            .ToObservable()
+            .SubscribeAwait(
+                async (changes, _) => { await UpdateOrderCache(); },
+                awaitOperation: AwaitOperation.Sequential
+            )
+            .AddTo(_disposables);
+    }
+
+
+    public async Task SetRelativePosition(ISortableItem sortableItem, int delta)
+    {
+        await _semaphore.WaitAsync();
+        try
+        {
+            var redModSortableItem = (RedModSortableItem)sortableItem;
+            // Get a stagingList of the items in the order
+            var stagingList = _orderCache.Items
+                .OrderBy(item => item.SortIndex)
+                .ToList();
+
+            // Get the current index of the item relative to the full list
+            var currentIndex = stagingList.IndexOf(sortableItem);
+
+            // Get the new index of the group relative to the full list
+            var newIndex = currentIndex + delta;
+
+            // Ensure the new index is within the bounds of the list
+            if (newIndex < 0 || newIndex >= stagingList.Count)
+                return;
```

Comment:
We have an unused return value here; think it might be worthwhile to return error cases as a result?
We could also possibly clamp the delta to min/max. Just some random thoughts that came to mind.